### PR TITLE
Fix: Validate class_weight keys in model.fit

### LINKED ISSUE #23220: Invalid class_weight keys silently ignored in model.fit(), no error raised
When passing an invalid `class_weight` dictionary (with a key that doesn't match any class index) into `model.fit()`, no error or warning is raised. Training proceeds silently as if `class_weight` was not provided, which can hide real bugs in user code.

**Reproduction:**

import tensorflow as tf
from tensorflow.keras import layers

class MyModel(tf.keras.Model):
    def __init__(self):
        super().__init__()
        self.dense = layers.Dense(4, activation='relu')

    def call(self, inputs):
        return self.dense(inputs)

model = MyModel()
model.compile(optimizer='adam', loss='mse')

x = tf.random.normal((10, 1))
y = tf.ones((10, 1))

model.fit(x, y, class_weight={'invalid': 0.5}, verbose=0)
# No error raised — expected ValueError or KeyError

**Expected behavior:**
Invalid class_weight keys should raise a `ValueError` or `KeyError`.

**Context:**
Originally reported at tensorflow/tensorflow#98283. A fix was attempted at tensorflow/tensorflow#98429, but the maintainer closed it noting that class_weight validation logic should live in the Keras repo instead of the legacy tf.keras path in TensorFlow core.

### PR DESCRIPTION
**Pull request by Hasnaathussain**
This PR introduces validation logic to ensure that keys passed to `class_weight` in `model.fit()` are valid class indices present in the target labels, raising a clear `ValueError` for invalid keys instead of silently ignoring them. Fixes #23220.

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

**Comment by Hasnaathussain** [comment]
Hi @gbaned - This PR has been ready for review for 1 day and all CI checks are passing.

**Summary:**
- Fixes issue #23220: Validates class_weight keys to ensure they match target labels
- Prevents silent ignoring of invalid class weights in model.fit()
- Comprehensive test coverage included
- All CI checks passing

**Status:** ✅ Ready for review and merge

Could you please review and approve when ready?

**Comment by hertschuh** [comment]
This was fixed by https://github.com/keras-team/keras/pull/23221
